core/workflow-core/src/main/scala/edu/uci/ics/amber/core/storage/result/iceberg/IcebergTableManager.scala

```diff
@@ -0,0 +1,213 @@
+package edu.uci.ics.amber.core.storage.result.iceberg
+
+import edu.uci.ics.amber.util.IcebergUtil
+import org.apache.iceberg.Table
+import scala.collection.mutable
+import org.apache.iceberg.types.Types
+import java.time.ZoneId
+import java.util.Date
+import edu.uci.ics.amber.core.storage.IcebergCatalogInstance
+import scala.jdk.CollectionConverters._
+import org.apache.iceberg.data.Record
+
+/**
+  * IcebergTableManager provides utility methods for calculating statistics
+  * (e.g., numeric, date, categorical) for an Iceberg table.
+  *
+  * @param table The Iceberg table managed by this class.
+  */
+class IcebergTableManager(table: Table) {
```

Comment:
Can you define a public class called `ResultTableStatistics`, which is a collection of the stats related to the `ResultTable`. 

This `ResultTableStatistics` will have the `getNumericColStats`, `getDateColStats`, and `getCategoricalStats` method. 

You can add another function to the `IcebergUtil`, possibly called `getTableStatistics` that creates the `ResultTableStatistics` and returns it.

The way of computing `ResultTableStatistics` is to use the APIs like `valueCounts`, `lowerBounds` and `upperBound`. And you can use `table.schema()` to decide what are the date columns, categorical columns and numerical columns.